source/elements/oneDAL/source/data_management/index.rst

```diff
@@ -5,12 +5,293 @@
+============
+
+|dal_short_name| provides a set of concepts to operate on out-of-memory and
+in-memory data during different stages of the :ref:`data analytics pipeline
+<data_analytics_pipeline>`.
+
+.. _dataset:
+
+Dataset
+--------
+
+The most common concept over the data in |dal_short_name| is a :term:`dataset`.
+This is an in-memory or out-of-memory tabular view of data, where table rows
+represent :term:`observations <Observation>` and columns represent
+:term:`features <Feature>`.
+
+.. image:: _static/dataset.png
+  :width: 400
+  :alt: Dataset
+
+The dataset comes across all stages of the data analytics pipeline. For example:
+
+1. On the acquisition stage, it is downloaded into the
+   local memory
+
+2. On the preparation stage, it is converted to numerical
+   representation
+
+3. On the computation stage, it is used as one of the :ref:`inputs <Input>` or
+   :ref:`results <Result>` of an algorithm or even can be found in its
+   parameters
+
+.. _data-source:
+
+Data source
+-----------
+
+Data source is a concept of an out-of-memory storage of a :term:`dataset`. It is
+used on data acquisition and data preparation stages for:
+
+- Dataset extraction from databases, files, remote storages, etc.
+
+- Dataset loading into the local memory of the device. Data are not always fit
+  the local memory, especially when processing with accelerators. Data source
+  provides an ability to load data by batches and orchestrates its efficient
+  extraction directly into local memory of the device. Therefore data source
+  enables complex scenarios of data analytics, like :term:`online computations
+  <Online mode>`.
+
+- Filtering or normalization of :term:`feature` values being extracted.
+
+- Recovery (emulation) of missing :term:`feature` values.
+
+- :term:`outlier` detection and recovery of the abnormal data.
+
+- Dataset transformation to numerical representation. Data sources can
+  automatically transform non-numeric :term:`categorical <Categorical feature>`
+  and :term:`continuous <Continuous feature>` values of data into one of numeric
+  :term:`data formats <Data format>`.
+
+.. _table:
+
+Table
+-----
+
+Table is a concept of a :term:`dataset` with in-memory numerical data. It is
+used on data-preparation and data-processing stages to:
+
+- Store heterogeneous (in the general case) in-memory data with variety of
+  :term:`data formats <Data format>` (dense, sparse, chunked, contiguous, etc.)
+
+- Avoid unnecessary data copies on conversion from external data
+  representations.
+
+- Transfer memory ownership of the data from user responsibility to the table,
+  or share it between them
+
+- Connect with :ref:`data-source` to convert out-of-memory dataset
+  representation to in-memory view
+
+- Stream the data to the algorithm
+
+- Access the underlying data on a device in a required :term:`data format`, e.g.
+  by blocks of defined :term:`data layout`.
+
+Table provide read-only access to the data inside it for thread-safety reasons
```

Comment:
```suggestion
A table provides a read-only access to the data within it for thread-safety reasons
```

i'd also switch the order: `for thread-safety reasons...., a table provides ...`